internal/backend/oauth/webhook.go

```diff
@@ -85,7 +85,7 @@ func (s *svc) exchange(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// Ensure the state parameter is well-formed.
-	sub := regexp.MustCompile(`^(.+)_([a-z]+)$`).FindStringSubmatch(state)
```

Comment:
The change from `+` to `*` is incorrect because the `state` param is supposed to be an AK ID.